rcl_logging_interface/include/rcl_logging_interface/rcl_logging_interface.h

```diff
@@ -91,6 +95,59 @@ RCL_LOGGING_INTERFACE_PUBLIC
 RCUTILS_WARN_UNUSED
 rcl_logging_ret_t rcl_logging_external_set_logger_level(const char * name, int level);
 
+/// Get the logging directory.
+/**
+ * Uses various environment variables to construct a logging directory path.
+ *
+ * Use $ROS2_LOG_DIR if ROS2_LOG_DIR is set and not empty.
+ * Otherwise, use $ROS2_HOME/log, using ~/.ros for ROS2_HOME if not set or if empty.
+ *
+ * It also expands '~' to the current user's home directory.
+ *
+ * \param[inout] directory The C string pointer at which to write the directory path.
+ *   Only meaningful if the call is successful.
+ * \param[in] allocator The allocator to use for memory allocation.
```

Comment:
In some places we pass the allocator by value, in some others by pointer.
More often, we do the second so I would change the signature to `const rcutils_allocator_t * allocator)`.

PS: We should really have an item in the style guide clarifying that.

Reply:
yeah, I was wondering which way was the preferred one! Changed in 0705dd3a57a3e36594ac9a2d054a1b54a715fef8

Reply:
> PS: We should really have an item in the style guide clarifying that.

here? https://index.ros.org/doc/ros2/Contributing/Code-Style-Language-Versions/

I can open an issue (and do the change too)

Reply:
> I can open an issue (and do the change too)

That would be great, thanks!

> yeah, I was wondering which way was the preferred one! Changed in 0705dd3a57a3e36594ac9a2d054a1b54a715fef8

The fix looks good!

Reply:
https://github.com/ros2/ros2_documentation/issues/880

Reply:
Wait why are we passing by pointer?

Passing by pointer requires another check in each function that takes the pointer (check for nullptr). Since it's never ok for the allocator to be nullptr and since copying the allocator is trivial, I'd say pass by value.

Reply:
> Passing by pointer requires another check in each function that takes the pointer (check for nullptr). Since it's never ok for the allocator to be nullptr and since copying the allocator is trivial, I'd say pass by value.

The rationale sounds good to me, but I've seen a lot of functions in our API taking a pointer.

Reply:
Yeah, and I've probably written several of them, but if I had to be pressed on which to prefer, I'd say by value unless you have a reason not to do so (like if the allocator is optional or there is a performance reason).

Reply:
alright, switched back to passing by value in 2a519a886c220f1619fcd239d34999ca2668c4b8